src/main/java/org/myrobotlab/service/ServoMixer.java

```diff
@@ -175,305 +256,345 @@ public Pose loadPose(String name) {
+
+      String filename = new File(c.posesDir).getAbsolutePath() + File.separator + name + ".yml";
+      log.info("loading pose name {}", filename);
+      String yml = FileIO.toString(filename);
+      return CodecUtils.fromYaml(yml, Pose.class);
+      // pose = Pose.loadPose(filename);
       // broadcastState(); "maybe too chatty"
     } catch (Exception e) {
       error(e);
     }
-    return pose;
+    return null;
   }
 
-  public void removePose(String name) {
-    try {
-      if (!checkDir(servoMixerDirectory)) {
-        error("invalid poses directory %s", servoMixerDirectory);
-        return;
+  public List<String> getPoseFiles() {
+    ServoMixerConfig c = (ServoMixerConfig)config;
+
+    List<String> files = new ArrayList<>();
+    if (!FileIO.checkDir(c.posesDir)) {
+      return files;
+    }
+
+    File dir = new File(c.posesDir);
+
+    if (!dir.exists() || !dir.isDirectory()) {
+      error("%s not a valid directory", c.posesDir);
+      return files;
+    }
+    File[] all = dir.listFiles();
+    Set<String> sorted = new TreeSet<>();
+    for (File f : all) {
+      if (f.getName().toLowerCase().endsWith(".yml")) {
+        sorted.add(f.getName().substring(0, f.getName().lastIndexOf(".")));
       }
-      String filename = new File(servoMixerDirectory).getAbsolutePath() + File.separator + name + ".pose";
-      File del = new File(filename);
-      if (del.exists()) {
-        del.delete();
+    }
+    for (String s : sorted) {
+      files.add(s);
+    }
+    return files;
+  }
+
+  public String getPosesDirectory() {
+    ServoMixerConfig c = (ServoMixerConfig)config;
+    return c.posesDir;
+  }
+
+  public List<ServoControl> listAllServos() {
+    ArrayList<ServoControl> servos = new ArrayList<ServoControl>();
+    for (ServiceInterface service : Runtime.getServices()) {
+      if (service instanceof ServoControl) {
+        servos.add((ServoControl) service);
       }
-      invoke("getPoseFiles");
-    } catch (Exception e) {
-      error(e);
     }
+    return servos;
+  }
+
+  public void moveToPose(String name) throws IOException {
+    Pose p = getPose(name);
+    if (p == null) {
+      error("cannot find pose %s", name);
+    }
+    moveToPose(name, p, false);
   }
 
-  public void removeSequence(String name) {
+  public void moveToPose(String name, Pose p, boolean blocking) {
     try {
-      if (!checkDir(servoMixerDirectory)) {
-        error("invalid poses directory %s", servoMixerDirectory);
+      
+      if (p.moves == null) {
+        error("no moves within pose file %s", name);
         return;
       }
-      String filename = new File(servoMixerDirectory).getAbsolutePath() + File.separator + name + ".seq";
-      File del = new File(filename);
-      if (del.exists()) {
-        del.delete();
+      
+      for (String sc : p.moves.keySet()) {
+        PoseMove pm = p.moves.get(sc);
+        ServoControl servo = (ServoControl) Runtime.getService(sc);

```

Comment:
Potential ClassCastException here if `sc` was created as a different service. Mainly a reminder for me to add a new overload to getService() that takes the expected service type

Reply:
it should be an interface ServoControl not service type.. at least that would give someone the flexibility to use DIYServo or potentially something besides a servo which implements the same interface.
For the most access patterns Runtime.getService "should" be return interfaces only

Reply:
Sorry, my comment wasn't clear enough. I meant if, for example, the pose moves a servo called "test". But there's a service called "test" that is instead a ProgramAB service for example. Usually shouldn't happen, but if it does we should probably fail gracefully instead of throwing an arcane exception